### PR TITLE
Bitbucket: start refactoring api versions

### PR DESCRIPTION
**Pull request by djachkov**
Start removing messy api versions

### CODE REVIEW

File: atlassian/rest_client.py [hunk 1]
```diff
@@ -201,15 +201,15 @@ def get(self, path, data=None, flags=None, params=None, headers=None, not_json_r
         """
         response = self.request('GET', path=path, flags=flags, params=params, data=data, headers=headers,
                                 trailing=trailing)
-        if self.advanced_mode:
-            return response
         if not_json_response:
             return response.content
+        if self.advanced_mode:
+            return response
```
Comment (djachkov) [inline atlassian/rest_client.py:207]: Please pay attention!
In this place, the order of lines was messed and the 'else' block below ran in wrong cases

Comment (gonchik) [reply atlassian/rest_client.py:207]: Could you reverse that change, please?

File: atlassian/rest_client.py [hunk 2]
```diff
@@ -201,15 +201,15 @@ def get(self, path, data=None, flags=None, params=None, headers=None, not_json_r
...
         else:
             if not response.text:
                 return None
             try:
-                return response.json()
+                return response.json() or {}
```
Comment (gonchik) [inline atlassian/rest_client.py:212]: That one is excellent!

File: atlassian/bitbucket.py [hunk 3]
```diff
@@ -14,32 +14,34 @@ def __init__(self, *args, **kwargs):
         super(Bitbucket, self).__init__(*args, **kwargs)
         url = kwargs.pop('url', False)
```
Comment (Spacetown) [inline atlassian/bitbucket.py:15]: This blck should be in front of the base init. See #584.

File: atlassian/bitbucket.py [hunk 4]
```diff
@@ -139,16 +133,16 @@ def project_keys(self, key, limit=99999, filter_str=None):
         :param filter_str:  OPTIONAL: users filter string
         :return:
         """
-        if not self.cloud:
-            url = 'rest/keys/1.0/projects/{key}/ssh'.format(key=key)
-        else:
-            url = 'rest/keys/2.0/projects/{key}/ssh'.format(key=key)
+        resource = self.resource_url('projects/{key}/ssh'.format(key=key))
```
Comment (Spacetown) [inline atlassian/bitbucket.py:136]: Here ressource_url MUST not be used, url is .../rest/keys/... and not .../rest/api/...

File: atlassian/bitbucket.py [hunk 5]
```diff
@@ -183,30 +176,32 @@ def repo_keys(self, project_key, repo_key, limit=99999, filter_str=None):
         :param filter_str:  OPTIONAL: users filter string
         :return:
         """
-        if not self.cloud:
-            url = 'rest/keys/1.0/projects/{project_key}/repos/{repo_key}/ssh'.format(project_key=project_key,
-                                                                                     repo_key=repo_key)
-        else:
-            url = 'rest/keys/2.0/projects/{project_key}/repos/{repo_key}/ssh'.format(project_key=project_key,
-                                                                                     repo_key=repo_key)
+        resource = self.resource_url('projects/{project_key}/repos/{repo_key}/ssh'.format(project_key=project_key,
```
Comment (Spacetown) [inline atlassian/bitbucket.py:179]: Also .../rest/keys/... and not .../rest/api/...

**Review by Spacetown — CHANGES_REQUESTED** [review]
This change was also my intension after #584 .

File: atlassian/bitbucket.py (hunk 5 above)
Comment (gonchik) [reply atlassian/bitbucket.py:179]: @Spacetown Any good idea around that?

Comment (djachkov) [reply atlassian/bitbucket.py:179]: thx, I just missed this part!

Comment (Spacetown) [reply atlassian/bitbucket.py:179]: Create a member function ressource_url_keys which does mealy the same as the ressource_url. I'll create a PR for this.

**Pull request closed without merging**